### ISSUE TITLE
Skull trap visual bugs

### ISSUE THREAD

**Issue by MarcelHB** [gemrb/gemrb#2047]
#### Bug description
When casting skull trap (in IWD2: SPWI313Z), you notice two things:

1. Some bad rendering within close range of the trap, it's misplaced sparkle sprites or something that moves together with the camera.
2. When exploding, there should be a scattering of fragments around – that's missing. Potentially subject to a high fps bug, I checked that on 30fps, too.

#### Steps to reproduce
1. Cast said spell
2. Try IWD2, if bug 1. is not noticable anywhere else.

#### Expected behavior
No visual bugs, and scattering as seen [here](https://www.youtube.com/watch?v=u5lhH8KhPCo) at 5:23.

#### Screenshots
![2024-03-03_01-36](https://github.com/gemrb/gemrb/assets/238558/a41391ac-210a-400d-8484-72d0cf5a8aac)

GemRB version (check as many as you know apply)
- [x] master as of this issue
- [ ] 0.9.2
- [ ] 0.9.1
- [ ] 0.9.0

Video Driver (check as many as you know apply)
- [ ] SDL1.2
- [x] SDL2 built with `OPENGL_BACKEND` enabled
- [ ] SDL2 without `OPENGL_BACKEND` enabled

**Comment by lynxlynxlynx** [comment]
2. is #128, so we can ignore it here.

**Comment by lynxlynxlynx** [comment]
I tried casting it and can't see the issue in 1.. Do you only get it in that area?

**Comment by lynxlynxlynx** [comment]
Was this testing done with your PR applied? Could be a regression from that, since the one problematic place is about triggered projectiles.

**Comment by MarcelHB** [comment]
> Was this testing done with your PR #2045 applied? Could be a regression from that, since the one problematic place is about triggered projectiles.

I can replay this on master, and it's reproducible but under unknown conditions, my best guess it's that you need a small map to actually recognize this bug. But that's just an impression from testing through some IWD2 save games, so on larger maps I haven't spotted anything.

**Comment by lynxlynxlynx** [comment]
Ah, I see now — for me they slowly drop from the top of the screen when positioned like in the screenshot. But fade eventually, it's odd that you managed to accumulate them.

What is it? I bet it's the exploded chunks and the positioning bug being in our `gemrb/unhardcoded/iwd2/trapskul.pro`. The rest is still the same old bug.

**Comment by bradallred** [comment]
The bad rendering (at least the blending) is clearly just a missing application of `ONE_MINUS_DST`. This could either be the projectile missing `PTF_TRANS` or that it also has `PTF_BLEND`. You should be able to fix it by tweaking the flags on the pro.

**Comment by lynxlynxlynx** [comment]
I doubt it should be there at all, I don't see anything interesting in the pro. SKLT is the head and a shadow bam is attached. SKLH has the parts for the explosion and that's what we don't draw properly. So perhaps there's also a desync with the explosion. We already look it up, so this must be it, I just don't see from where it comes — it will require more than code reading.

**Comment by bradallred** [comment]
something doesn't add up. `SKLT` and `SKLH` don't have a black background, but the rendering in the screenshot shows one. Are we palette munging them for some reason?

**Comment by lynxlynxlynx** [comment]
Yes, this is the `SP_TYPE_BITMAP` type of sparkles, triggered in `Projectile::DrawExplosion`. So yeah, if it's not a duplicate of #128, it's very close to it.
